airflow/gcp/operators/display_video.py

```diff
@@ -0,0 +1,301 @@
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    @staticmethod
+    def _set_report_name(name: str) -> str:
+        return name if name.endswith(".csv") else name + ".csv"
+
+    @staticmethod
+    def _set_bucket_name(name: str) -> str:
+        bucket = name if not name.startswith("gs://") else name[5:]
+        return bucket.strip("/")
+
+    def execute(self, context: Dict):
+        hook = GoogleDisplayVideo360Hook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version=self.api_version,
+        )
+        gcs_hook = GoogleCloudStorageHook(
+            google_cloud_storage_conn_id=self.gcp_conn_id, delegate_to=self.delegate_to
+        )
+
+        self.log.info("Starting downloading report %s", self.report_id)
+        resource = hook.get_query(query_id=self.report_id)
+        file_url = resource["metadata"]["googleCloudStoragePathForLatestReport"]
+
+        # Download the report
+        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+            with urllib.request.urlopen(file_url) as response:
+                shutil.copyfileobj(response, temp_file)
+
+            temp_file.flush()
+            # Upload the local file to bucket
+            gcs_hook.upload(
+                bucket_name=self.bucket_name,
+                object_name=self.report_name,
+                gzip=self.gzip,
+                filename=temp_file.name,
+                mime_type="text/csv",
+            )
+        self.log.info(
+            "Report %s was saved in bucket %s as %s.",
+            self.report_id,
+            self.bucket_name,
+            self.report_name,
+        )
+
+
+class GoogleDisplayVideo360RunReportOperator(BaseOperator):
+    """
+    Runs a stored query to generate a report.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleDisplayVideo360RunReportOperator`
+
+    .. seealso::
+        Check also the official API docs:
+        `https://developers.google.com/bid-manager/v1/queries/runquery`
+
+    :param report_id: Report ID to run.
+    :type report_id: str
+    :param params: Parameters for running a report as described here:
+        https://developers.google.com/bid-manager/v1/queries/runquery
+    :type params: Dict[str, Any]
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    template_fields = ("report_id", "params")
+
+    @apply_defaults
+    def __init__(
+        self,
+        report_id: str,
+        params: Dict[str, Any],
```

Comment:
Could we rename this (e.g. "body") as params is typically reserved for Jinja templating?